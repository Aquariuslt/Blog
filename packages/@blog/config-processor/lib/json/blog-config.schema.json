{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "http://github.com/aquariuslt/blog/config.schema.json",
  "title": "Blog Configuration",
  "description": "blog config file yml specification",
  "type": "object",
  "properties": {
    "version": {
      "type": "string",
      "default": "1.0.0"
    },
    "site": {
      "$ref": "#/definitions/site"
    },
    "build": {
      "$ref": "#/definitions/build"
    },
    "features": {
      "$ref": "#/definitions/features"
    }
  },
  "required": ["version", "site", "build"],
  "definitions": {
    "site": {
      "type": "object",
      "properties": {
        "host": {
          "type": "string",
          "format": "url",
          "description": "Your host, e.g. http://your-site.com or https://my-site.com."
        },
        "title": {
          "type": "string",
          "description": "Your blog title."
        },
        "author": {
          "type": "string",
          "description": "Your name"
        },
        "avatar": {
          "type": "string",
          "description": "Your profile picture"
        },
        "subtitle": {
          "type": "string",
          "description": "Your blog sub title"
        },
        "description": {
          "type": "string",
          "description": "Your blog description, this usually will built in html head meta"
        },
        "language": {
          "type": "string",
          "description": "Your prefer language",
          "enum": ["zh", "en"],
          "default": "zh"
        },
        "copyright": {
          "type": "string",
          "description": "Site copyright, usually appear under your website",
          "default": "copyright 2019 aquariuslt blog"
        }
      },
      "required": ["host", "title", "subtitle", "description"]
    },
    "build": {
      "type": "object",
      "properties": {
        "directory": {
          "type": "object",
          "properties": {
            "posts": {
              "type": "string",
              "description": "markdown posts file base directory, relative to project root path",
              "default": "posts"
            },
            "public": {
              "type": "string",
              "description": "build location path",
              "default": "dist"
            }
          },
          "required": []
        },
        "favicon": {
          "type": "string",
          "description": "Your favicon file location",
          "default": "favicon.ico"
        },
        "theme": {
          "type": "string",
          "description": "Your blog theme package name, not implementation now",
          "default": "vue"
        },
        "colors": {
          "type": "object",
          "description": "Your blog theme color preset from material color customization",
          "default": {}
        }
      },
      "required": []
    },
    "features": {
      "type": "object",
      "properties": {
        "ga": {
          "type": "string",
          "description": "Your Google Analytics tracking id"
        },
        "disqus": {
          "type": "string",
          "description": "Your disqus short id"
        }
      },
      "required": []
    }
  }
}
