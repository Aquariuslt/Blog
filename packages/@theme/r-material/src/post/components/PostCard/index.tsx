import * as React from 'react';
import { createStyles, makeStyles, StyleRules ,Theme } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import Button from '@material-ui/core/Button';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CardActionArea from '@material-ui/core/CardActionArea';

import styles from '../../../styles';

type PostCardProps = BlogModel.Post;


const useStyles = makeStyles(
  (theme: Theme): StyleRules =>
    createStyles({
      root: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-around',
        overflow: 'hidden',
        backgroundColor: theme.palette.background.paper,
        paddingBottom: 8
      },
      card: {
        maxWidth: styles.content.maxWidth,
        marginLeft: 8,
        marginRight: 8
      },
      icon: {
        color: 'rgba(255, 255, 255, 0.54)'
      }
    })
);

const POST_URL_PREFIX = '/posts';

const PostCard: React.ComponentType<PostCardProps> = (props: PostCardProps): JSX.Element => {
  const classes = useStyles();

  return (
    <Grid
      container
      className={classes.root}
    >
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        lg={12}
        xl={12}
        className={classes.root}
      >
        <Card className={classes.card}>
          <CardActionArea
            href=""
          >
            {
              props.metadata.cover &&
              <CardMedia
                component="img"
                alt={props.metadata.title}
                height="140"
                image={props.metadata.cover}
                title={props.metadata.title}
              />
            }
            <CardContent>
              <Typography gutterBottom variant="h5" component="h2">
                {props.metadata.title}
              </Typography>
              <Typography variant="body2" color="textSecondary" component="p">
                {props.summary}
              </Typography>
            </CardContent>
          </CardActionArea>
          <CardActions>
            <Button href={POST_URL_PREFIX + props.permalink} size="small">
              More
            </Button>
          </CardActions>
        </Card>
      </Grid>
    </Grid>
  );
};

export default PostCard;
