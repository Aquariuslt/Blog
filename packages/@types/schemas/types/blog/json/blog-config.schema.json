{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "http://github.com/aquariuslt/blog/config.schema.json",
  "description": "blog config file yml specification",
  "type": "object",
  "properties": {
    "version": {
      "type": "string",
      "default": "1.0.0"
    },
    "site": {
      "$ref": "#/definitions/site"
    },
    "build": {
      "$ref": "#/definitions/build"
    }
  },
  "required": [
    "version",
    "site",
    "build"
  ],
  "definitions": {
    "site": {
      "type": "object",
      "properties": {
        "host": {
          "type": "string",
          "format": "url"
        },
        "title": {
          "type": "string"
        },
        "subtitle": {
          "type": "string"
        },
        "description": {
          "type": "string"
        },
        "language": {
          "type": "string",
          "enum": [
            "zh",
            "en"
          ],
          "default": "zh"
        },
        "copyright": {
          "type": "string",
          "default": "copyright 2019 aquariuslt blog"
        }
      },
      "required": [
        "host",
        "title",
        "subtitle",
        "description"
      ]
    },
    "build": {
      "type": "object",
      "properties": {
        "directory": {
          "type": "object",
          "properties": {
            "posts": {
              "type": "string",
              "description": "markdown posts file base directory, relative to project root path",
              "default": "posts"
            },
            "public": {
              "type": "string",
              "description": "build location path",
              "default": "dist"
            }
          },
          "required": []
        },
        "favicon": {
          "type": "string",
          "default": "favicon.ico"
        },
        "theme": {
          "type": "string",
          "default": "vue"
        }
      },
      "required": []
    }
  }
}
