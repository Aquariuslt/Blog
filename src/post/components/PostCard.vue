<template>
  <md-card v-if="post" class="post-card">
    <post-header
      :metadata="post.metadata">
    </post-header>
    <post-catalogue
      :catalogue="post.metadata.catalogue">
    </post-catalogue>
    <post-content
      :content="post.html">
    </post-content>
    <post-footer
      :metadata="post.metadata">
    </post-footer>
  </md-card>
</template>

<script>
  import PostHeader from '@/post/components/PostHeader';
  import PostContent from '@/post/components/PostContent';
  import PostFooter from '@/post/components/PostFooter';
  import PostCatalogue from '@/post/components/PostCatalogue';

  export default {
    components: {
      PostCatalogue,
      PostFooter,
      PostContent,
      PostHeader
    },
    name: 'post-card',
    props: {
      post: Object
    }
  };
</script>

<style scoped>
  .post-card {
    width: 100%;
    max-width: 1084px;
    margin-left: 10px;
    margin-right: 10px;
  }
</style>

