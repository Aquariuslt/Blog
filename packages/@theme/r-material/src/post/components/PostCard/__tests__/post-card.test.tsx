import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import PostCard from '../';

import * as samplePost from './__mocks__/sample-post.json';

Enzyme.configure({adapter: new Adapter()});

describe('@theme/r-material: post/component: PostCard', (): void => {

  it('# should mount correctly', (): void => {

    const wrapper = Enzyme.shallow(<PostCard {...samplePost} />);
    expect(wrapper).toMatchSnapshot();
  })
});
