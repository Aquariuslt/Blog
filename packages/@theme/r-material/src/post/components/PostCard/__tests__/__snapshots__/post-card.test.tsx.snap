// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: post/component: PostCard # should mount correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-root-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-root-1"
    item={true}
    lg={12}
    md={12}
    sm={12}
    xl={12}
    xs={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-card-2"
    >
      <WithStyles(ForwardRef(CardActionArea))
        href=""
      >
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(Typography))
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Spring 整合 Apache Shiro 实现各等级的权限管理
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            component="p"
            variant="body2"
          >
            ## Background
前几个月在做一个常规的权限隔离功能的时候,恰好使用过Apache Shiro.
Apache Shiro 是一款Java的安全框架,通常用作Web应用的权限校验,身份验证.
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(CardActionArea))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          href="/posts/2015/10/25/apache-shiro-spring-integration"
          size="small"
        >
          More
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
